### PR TITLE
Add Gatsby gitignore

### PR DESCRIPTION
**Pull request by affeldt28**
# Pull Request

Thank you for contributing to @toptal/gitignore and https://www.gitignore.io.

## New or update

Select the appropriate check box for this pull request. This helps when merging to ensure there are no conflicts with other templates or misunderstandings of how thee template list works.

### New

- [x] Template - New `.gitignore` template
- [ ] Composition - Template made from smaller templates
- [ ] Inheritance - Template similar to an existing template
- [ ] Patch - Template extending functionality of existing template

### Update

- [ ] Template - Update existing `.gitignore` template

## Details

Create .gitignore template for [Gatsby](https://www.gatsbyjs.com/)

### CODE REVIEW

File: templates/Gatsby.gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+node_modules/
+
+.cache/
+public
+
+src/gatsby-types.d.ts
```
Comment (deniskaber) [inline templates/Gatsby.gitignore:6]: This seems to be specific to some project. I didn't find any confirmation that there is a convention to put gatsby types like here.

**Review by deniskaber — CHANGES_REQUESTED** [review]
Hey @affeldt28 ! Thank you for your contribution. I added a comment regarding your PR. Please take a look.

File: templates/Gatsby.gitignore (hunk 1 above)
Comment (affeldt28) [reply templates/Gatsby.gitignore:6]: This from the default gitignore when using the gatsby cli to create a new project and selecting typescript as primary language. So I would say its ok to add it here.

See here: https://github.com/gatsbyjs/gatsby/blob/master/starters/gatsby-starter-minimal-ts/.gitignore

**Review by filipechagas — APPROVED** [review]

File: templates/Gatsby.gitignore (hunk 1 above)
Comment (deniskaber) [reply templates/Gatsby.gitignore:6]: Ok, that makes sense! Thank you for providing this example.

**Review by deniskaber — APPROVED** [review]

**Pull request merged**